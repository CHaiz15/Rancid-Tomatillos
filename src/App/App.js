import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import './App.scss';
import MoviesPage from '../MoviesPage/MoviesPage.js';
import Login from '../Login/Login'
import MovieDetails from '../MovieDetails/MovieDetails.js'
import { loadMovies } from '../Actions'
import { getMovies } from '../apiCalls';

class App extends Component {
  constructor() {
    super();
    this.state = {
    }
  }
  componentDidMount() {
    getMovies()
      .then(movies => this.props.loadMovies(movies))
      .catch(err => err)
  }

  logout = () => {
    window.location.href = '/';
  }

  render() {
    return (
      <main className="App">
        <nav className='nav-bar'>
          <div className='logo'>
            <div className='film-icon'></div>
            <div>Rancid Tomas</div>
          </div>
          {this.props.user.name ? 
            <div>
              <h3>Welcome, {this.props.user.name}</h3>
              <button onClick={this.logout}>Logout</button>
            </div>:
            <NavLink className='login-btn' to='/login' type='button'>Login</NavLink>}
        </nav>
        <Route exact path='/login' component={Login} />
        <Route exact path='/' component={MoviesPage} />
        <Route path='/movies/:id' render={({ match }) => {
          const movie = this.props.movies.find(movie => parseInt(match.params.id) === movie.id)
          if (!movie) {
            return (<div>This movie does not exist! </div>);  
          }
          return <MovieDetails {...movie} />
        }} />
      </main>
    );
  }
}

const mapStateToProps = state => ({
  movies: state.movies,
  user: state.user,
})

const mapDispatchToProps = dispatch => ({
  loadMovies: movies => dispatch(loadMovies(movies))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
