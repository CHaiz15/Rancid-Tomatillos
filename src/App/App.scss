.nav-bar {
  display: flex;
  align-items: center;
  height: 5rem;
  border: black solid 1px;
  justify-content: space-between;
  padding: 0 1% 0 1%;
  position: fixed;
  overflow: hidden;
  background-color: #fff;
  width: 100%;
}

a {
  text-decoration: none;
  list-style: none;
}

.logo {
  display: flex;
  align-items: center;
}

.film-icon {
  background-image: url('../assets/film-icon.svg');
  height: 50px;
  width: 50px;
}

.login-btn {
  border: 1px black solid;
  padding: 1%;
  border-radius: 30px;
  font-size: 18px;
  width: 7%;
  text-align: center;
}
