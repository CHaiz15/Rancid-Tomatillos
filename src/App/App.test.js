import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { loadMovies } from '../actions';

describe('App', () => {
  let wrapper,mockState
  it.skip('should match the snapshot', () => {
    wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
  
  describe('mapStateToProps', () => {
    it('should return an object with movies and a user', () => {
      mockState = {
        movies: [{
          id: 21,
          title: "Sonic the Hedgehog",
          "poster_path": "https://image.tmdb.org/t/p/original//aQvJ5WPzZgYVDrxLX4R6cLJCEaQ.jpg",
          "backdrop_path": "https://image.tmdb.org/t/p/original//tCUcf3oNWMW8kwAj3WC6CvIN5ah.jpg",
          "release_date": "2020-02-12",
          "overview": "Based on the global blockbuster videogame franchise from Sega, Sonic the Hedgehog tells the story of the world’s speediest hedgehog as he embraces his new home on Earth. In this live-action adventure comedy, Sonic and his new best friend team up to defend the planet from the evil genius Dr. Robotnik and his plans for world domination.",
          "average_rating": 8
        }],
        user: {
          name: 'Ken'
        }
    }

      const expected = {
        movies: [{
          id: 21,
          title: "Sonic the Hedgehog",
          "poster_path": "https://image.tmdb.org/t/p/original//aQvJ5WPzZgYVDrxLX4R6cLJCEaQ.jpg",
          "backdrop_path": "https://image.tmdb.org/t/p/original//tCUcf3oNWMW8kwAj3WC6CvIN5ah.jpg",
          "release_date": "2020-02-12",
          "overview": "Based on the global blockbuster videogame franchise from Sega, Sonic the Hedgehog tells the story of the world’s speediest hedgehog as he embraces his new home on Earth. In this live-action adventure comedy, Sonic and his new best friend team up to defend the planet from the evil genius Dr. Robotnik and his plans for world domination.",
          "average_rating": 8
        }],
        user: {
          name: 'Ken'
        }
    }
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the dispatch(loadMovies) action when loadMovies(movies) is called', () => {
      const mockDispatch = jest.fn();
      const movies = [{
        id: 21,
        title: "Sonic the Hedgehog",
        "poster_path": "https://image.tmdb.org/t/p/original//aQvJ5WPzZgYVDrxLX4R6cLJCEaQ.jpg",
        "backdrop_path": "https://image.tmdb.org/t/p/original//tCUcf3oNWMW8kwAj3WC6CvIN5ah.jpg",
        "release_date": "2020-02-12",
        "overview": "Based on the global blockbuster videogame franchise from Sega, Sonic the Hedgehog tells the story of the world’s speediest hedgehog as he embraces his new home on Earth. In this live-action adventure comedy, Sonic and his new best friend team up to defend the planet from the evil genius Dr. Robotnik and his plans for world domination.",
        "average_rating": 8
      }];
      const actionToDispatch = loadMovies(movies);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.loadMovies(movies);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
});