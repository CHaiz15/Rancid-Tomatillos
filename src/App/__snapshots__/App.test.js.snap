// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="App"
>
  <nav
    className="nav-bar"
  >
    <Link
      className="logo"
      to="/"
    >
      <div
        className="film-icon"
      />
      <div>
        Rancid Tomas
      </div>
    </Link>
    <div
      className="user-nav"
    >
      <h3
        className="welcome"
      >
        Welcome, 
        Ken
      </h3>
      <button
        className="logout-btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </nav>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Login)",
        "type": [Function],
      }
    }
    exact={true}
    path="/login"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(MoviesPage)",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
  />
  <Route
    path="/movies/:id"
    render={[Function]}
  />
</main>
`;
